notebooks/use-cases/information-extraction.ipynb

```diff
@@ -0,0 +1,285 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "jfbjatzrbg",
+   "metadata": {},
+   "source": [
+    "## Installation\n",
+    "\n",
+    "First, we need to install the Docling package with VLM (Vision Language Model) support for information extraction:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8cf4340c-cfd4-418c-955b-be8c0d544e67",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from IPython import display\n",
+    "from pydantic import BaseModel, Field\n",
+    "from rich import print"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "99a1kzl8aw8",
+   "metadata": {},
+   "source": [
+    "## Setup and Configuration\n",
+    "\n",
+    "Let's import the necessary libraries and set up our environment:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6f5e2659-e626-4235-97fc-f311adf8f5b7",
+   "metadata": {},
+   "source": [
+    "### Sample Document\n",
+    "\n",
+    "For this demonstration, we'll work with a sample invoice document. This will help us understand how information extraction works with real-world documents:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f0daf568-dd36-4e07-9d61-f2a4f196c449",
+   "metadata": {
+    "editable": true,
+    "raw_mimetype": "",
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "invoice_url = \"https://slicedinvoices.com/pdf/wordpress-pdf-invoice-plugin-sample.pdf\"\n",
+    "\n",
+    "display.HTML(f'''\n",
+    "<iframe src=\"{invoice_url}\" width=\"100%\" height=\"600px\">\n",
+    "  <p>Your browser does not support iframes. <a href=\"{invoice_url}\">Click here to view the invoice</a></p>\n",
+    "</iframe>\n",
+    "''')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "20fa56da",
+   "metadata": {},
+   "source": [
+    "### Document Extractor Setup\n",
+    "\n",
+    "Now let's configure the document extractor to handle PDF and image formats. The extractor is the main component that will process our documents and extract information using the templates we define:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "46dd4e6e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from docling.datamodel.base_models import InputFormat\n",
+    "from docling.document_extractor import DocumentExtractor\n",
+    "\n",
+    "extractor = DocumentExtractor(allowed_formats=[InputFormat.IMAGE, InputFormat.PDF])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "extraction-templates",
+   "metadata": {},
+   "source": [
+    "## Information Extraction with Templates\n",
+    "\n",
+    "Docling supports different template formats for information extraction. Templates define the structure and data types of the information you want to extract from documents. Let's explore the different approaches:\n",
```

Comment:
If documentation about this exists in the Docling docs, maybe worth adding a link here.

Reply:
Added a link